File: Data Structures & Algorithms/three-integer-sum/submission-8.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> res = new ArrayList<>();
        int l,r;
        for(int i = 0;i<nums.length;i++){
            if(nums[i]>0){
                break;
            }
            if(i>0 && nums[i-1]==nums[i]){
                continue;
            }
            l = i+1;
            r = nums.length-1;
            while(l<r){
                int s = nums[i]+nums[l]+nums[r];
                if(s<0){
                    l++;
                } else if (s>0){
                    r--;
                } else{
                    res.add(Arrays.asList(nums[i],nums[l],nums[r]));
                    l++;
                    r--;
                    while (l<r && nums[l] == nums[l-1]){
                        l+=1;
                    }
                }

            }

        }
        return res;

        
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-8.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        l,r = 0,0
        N = len(nums)
        res = 0
        while True:
            if r == N-1:
                return res
            res +=1
            nextR = r
            for i in range(l,r+1,1):
                nextR = max(i+nums[i],nextR)
            if nextR>=N-1:
                return res
            l = r+1
            r = nextR
        




        

        